src/network/networkd-link.c

```diff
@@ -913,10 +917,9 @@ static int link_request_set_neighbors(Link *link) {
                 }
         }
 
-        if (link->neighbor_messages == 0) {
+        if (link->neighbor_messages == 0)
                 link->neighbors_configured = true;
-                link_check_ready(link);
```

Comment:
You shouldn't get rid of this in case we re-arrange the logic for setting neighbors. All of the other functions perform the same check which should be safe regardless of where it is executed.